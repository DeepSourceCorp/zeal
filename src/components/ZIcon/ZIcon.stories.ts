import ZIcon from './ZIcon.vue'

import '../../assets/css/tailwind.css'
import '../../assets/css/typography.css'
import '../../assets/css/layout.css'

export default {
  title: 'Icons',
  component: ZIcon,
  excludeStories: /.*Data$/
}

export const Default = () => ({
  components: { ZIcon },
  template: `<div class='wrapper'>
        <z-icon icon='arrow-up'></z-icon>
        <p class="icon-title">arrow-up</p>
    </div>`
})

export const IconWithCustomColor = () => ({
  components: { ZIcon },
  template: `<div class="wrapper">
        <z-icon icon="activity" color="juniper"></z-icon>
        <p class="icon-title">activity</p>
    </div>`
})

export const SmallIcon = () => ({
  components: { ZIcon },
  template: `<div class='wrapper'>
        <z-icon icon='arrow-up' size="small"></z-icon>
        <p class='icon-title'>arrow-up</p>
    </div>`
})
