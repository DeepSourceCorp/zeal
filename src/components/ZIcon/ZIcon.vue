<template>
    <svg 
      :class="[
        'z-icon',
        `${size && `z-icon--${size}`}`,
        `${icon && `z-icon--${icon}`}`
      ]"
      viewBox="0 0 24 24" 
      v-html="getIcon(icon)"></svg>
</template>

<script lang="ts">
import { Component, Prop, Vue } from 'vue-property-decorator'
import feather from 'feather-icons'
import customIcons from '../../utils/icon-helper'

@Component({
  name: 'ZIcon'
})
export default class Icon extends Vue {
  @Prop({ default: '' })
  private icon!: string
  @Prop({default: ''})
  private size!: string

  public getIcon(iconName: string): HTMLOrSVGImageElement {
    let DOM = null
    if (feather.icons[iconName]) {
      DOM = feather.icons[iconName] && feather.icons[iconName].contents
    } else {
      DOM = customIcons[iconName] &&  customIcons[iconName].contents;
    }
    return DOM
  }
}
</script>

<style lang="css" scoped>
  .z-icon {
    @apply  w-6 h-6 fill-current stroke-current text-slate stroke-2;
  }

  .z-icon--small {
    @apply w-4 h-4;
  }
</style>
