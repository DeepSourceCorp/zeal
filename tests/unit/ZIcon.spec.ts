import ZIcon from '../../src/components/ZIcon'

import { shallowMount } from '@vue/test-utils'

describe('ZIcon', () => {
  it('renders an empty icon', () => {
    const wrapper = shallowMount(ZIcon)

    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders an icon when icon prop/name is passed', () => {
    const wrapper = shallowMount(ZIcon, {
      propsData: {
        icon: 'github'
      }
    })
    expect(wrapper.classes()).toContain(`z-icon--github`)
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('renders a small icon when size prop is passed', () => {
    const wrapper = shallowMount(ZIcon, {
      propsData: {
        size: 'small',
        icon: 'github'
      }
    })
    expect(wrapper.classes()).toContain(`z-icon--small`)
    expect(wrapper.html()).toMatchSnapshot()
  })
})
