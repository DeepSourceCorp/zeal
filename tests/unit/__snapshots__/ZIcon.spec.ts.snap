// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZIcon renders a small icon when size prop is passed 1`] = `
<svg viewBox="0 0 24 24" class="z-icon z-icon--small z-icon--github">
  <path d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"></path>
</svg>
`;

exports[`ZIcon renders an icon 1`] = `<svg viewBox="0 0 24 24" class="z-icon"></svg>`;

exports[`ZIcon renders an icon when icon prop/name is passed 1`] = `
<svg viewBox="0 0 24 24" class="z-icon z-icon--github">
  <path d="M9 19c-5 1.5-5-2.5-7-3m14 6v-3.87a3.37 3.37 0 0 0-.94-2.61c3.14-.35 6.44-1.54 6.44-7A5.44 5.44 0 0 0 20 4.77 5.07 5.07 0 0 0 19.91 1S18.73.65 16 2.48a13.38 13.38 0 0 0-7 0C6.27.65 5.09 1 5.09 1A5.07 5.07 0 0 0 5 4.77a5.44 5.44 0 0 0-1.5 3.78c0 5.42 3.3 6.61 6.44 7A3.37 3.37 0 0 0 9 18.13V22"></path>
</svg>
`;
